### DISCUSSION TITLE
How to override SSH command in v7.4?

### DISCUSSION THREAD

**Discussion by DonovanChan** [deployphp/deployer#3904] [Help needed] [answered]
Hi. I'm deploying to a cloud service that requires a proprietary command for connecting to the server. For example, `customUtil ssh` instead of `ssh`. Can I override the ssh command in v7.4 to achieve this?

I've tried this:

```
set('ssh_multiplexing', false);
set('ssh_path', 'customUtil ssh');
set('shell', 'customUtil ssh');
```

And it's returning:

```
❯ vendor/bin/dep -vvv deploy -vvv --log="$(pwd)/storage/logs/deployer.log" uat-cloud
task deploy:info
[uat-cloud] /Users/me/Library/Application Support/Herd/bin/php82 /Users/me/Projects/my-project/vendor/deployer/deployer/dep worker --port 52513 --task deploy:info --host uat-cloud --log /Users/me/Projects/my-project/storage/logs/deployer.log --decorated -vvv
[uat-cloud] ssh '-A' 'user@cloud-server' ': 312ea2c2a239ac495b5c; customUtil ssh'
[uat-cloud] run if [ -d /data/ci/service/.dep ]; then echo +yes; fi
[uat-cloud] user@cloud-server: Permission denied (publickey).
[uat-cloud]  error  in info.php on line 6:
[uat-cloud] exit code -1 (Unknown error)
done deploy:info 365ms
ERROR: Task deploy:info failed!
```

I also tried hacking the `dep` executable to figure things out but haven't been able to build a new version of it successfully.

**Comment by antonmedv** [comment, marked as answer]
Currently this part is not configurable.

See https://github.com/deployphp/deployer/blob/15ce6fed793c021fb2db0c87c8b8c4646b543503/src/Component/Ssh/Client.php#L65

But we can add a config option here as well.

**Reply by DonovanChan** [reply]
Thanks for the prompt response. I'd really appreciate the option. 
Do you have any documentation or tips on how I'd make a custom build for now?

**Reply by antonmedv** [reply]
> Do you have any documentation or tips on how I'd make a custom build for now?

build new phar with bin/build
